src/pyroboplan/ik/differential_ik.py

```diff
@@ -219,16 +235,21 @@ def solve(
 
                 # Gradient descent step
                 if not nullspace_components:
-                    q_cur += alpha * J.T @ np.linalg.solve(jjt, error)
+                    q_step = alpha * J.T @ np.linalg.solve(jjt, error)
                 else:
                     nullspace_term = sum(
                         [comp(self.model, q_cur) for comp in nullspace_components]
                     )
-                    q_cur += alpha * (
+                    q_step = alpha * (
                         J.T @ (np.linalg.solve(jjt, error - J @ (nullspace_term)))
                         + nullspace_term
                     )
 
+                # Zero out the values for the ignored indices before returning.
+                for idx in self.options.ignore_joint_indices:
+                    q_step[idx] = 0.0
```

Comment:
I'm sure it does, but it will probably be a bit more manual effort and probably takes away from the pedagogical goals of this library... which are already being severely violated with this PR haha